src/pri-dual-vec.md

```diff
@@ -0,0 +1,193 @@
+# Primal and Dual Vectors
+
+The finite element method uses vectors of data in a variety of ways and the
+differences can be subtle.  MFEM defines `GridFunction`, `LinearForm`, and
+`Vector` classes which help to distinguish the different roles that vectors of
+data can play.
+
+<center>**Graphical summary of Primal, Dual, DoF, True DoF vectors**</center>
+![](img/FEM-LA-Diagrams.png)
+
+## Primal Vectors
+
+The finite element method is based on the notion that a smooth function can be
+approximated by a sum of piece-wise functions called *basis functions*:
+$$f(\vec{x})\approx\sum_i f_i \phi_i(\vec{x}) \label{expan}$$
+The support of an individual basis function, $\;\phi_i(\vec{x})$, will either be
+a single zone or a collection of zones that share a common vertex, edge, or
+face.  The expansion coefficients, $\;f_i$, are linear functionals of the field
+being approximated, $\;f(\vec{x})$ in this case.  The $\;f_i$ could be as simple
+as values of the function at particular points, called interpolation points,
+e.g. $\;f_i=f(\vec{x}\_i)$ or they could be integrals of the field over
+submanifolds of the domain, e.g. $\;f_i = \int_{\Omega_i}f(\vec{x})d\vec{x}$.
+There are many possibilities but the expansion coefficients must be linear
+functionals of $\;f(\vec{x})$.  The expansion coefficients are often called
+*degrees of freedom*, or *DoFs* for short, though in certain cases they may
+not be actually independent.  We'll discuss this more in a later
+section on [True DoFs](pri-dual-vec.md#true-degree-of-freedom-vectors).
+
+Once the basis functions are defined, with some unique ordering, the expansion
+coefficients can be stored in a vector using the same order.  Such a vector of
+coefficients is called a *primal vector*.  The original function,
+$\;f(\vec{x})$, can then be approximated using \eqref{expan}.  In practice this
+requires not only the primal vector of coefficients but also knowledge of the
+mesh and the basis functions for each element of the mesh.  In MFEM these
+collections of information are combined into `GridFunction` objects (or
```

Comment:
Since `GridFunction` is not a standard term, we might want to say that "we call GridFunction the object representing a (piece-wise polynomial?) function belonging to a finite element approximation space."

Reply:
I think starting the sentence with "In MFEM..." conveys the same meaning as "we call..." but the end of this proposed sentence is a good addition.